gnome-image-installer/pages/diskimage/gis-diskimage-page.c

```diff
@@ -511,18 +517,24 @@ gis_diskimage_page_mount (GisPage *page)
       UDisksObject *object = UDISKS_OBJECT (l->data);
       UDisksBlock *block = udisks_object_peek_block (object);
       UDisksFilesystem *fs = NULL;
+      gboolean use_uuid = FALSE;
+      gboolean is_live = gis_store_is_live_install ();
       const gchar *const*mounts = NULL;
       const gchar *dev = NULL;
+      const gchar *uuid = gis_store_get_image_uuid ();
 
       if (block == NULL)
         continue;
 
       dev = udisks_block_get_preferred_device (block);
 
-      if (!g_str_equal (label, udisks_block_get_id_label (block)))
+      if (is_live && g_str_equal (uuid, udisks_block_get_id_uuid (block)))
+        use_uuid = TRUE;
+
+      if (!g_str_equal (label, udisks_block_get_id_label (block)) && !use_uuid)
         continue;
 
-      if (udisks_block_get_hint_ignore (block))
+      if (udisks_block_get_hint_ignore (block) && !use_uuid)
```

Comment:
I think this change would break the exFAT USB case, unfortunately. In that case there are two devices with the UUID from the kernel command line:

1. The real partition (eg `/dev/sdb1`). This partition cannot be used because it is mapped to a device-mapper device; attempts to mount it will fail. This one gets flagged as `UDISKS_IGNORE` in `eos-image-boot-generator`.
2. `/dev/mapper/endless-image-device`, another DM device mapped over the whole image host partition by `eos-image-boot-dm-setup` in `eos-boot-helper`. This one actually works, but probably doesn't exist in the `iso9660` case because only `ntfs` and `exfat` are handled in its copy-pasted logic to get the image file extents. 

So with this change, the ignore flag will be ignored. In the ISO case that will be fine, but in the exfat case it may (or may not, I don't think the ordering is defined) pick the real partition and fail to mount it.

I can see a few possible fixes (others may exist):

1. Drop the logic in `eos-image-boot-dm-setup` which gets the image file extents and punches holes in the mapped image
2. Keep that logic, but add an `iso9660` case which doesn't punch out any holes
3. Keep that logic, but only when the host device is read-write (there's no danger of overwriting your Endless image on a read-only device!)
4. Adjust `dracut/image-boot/eos-image-boot-generator` to not set `UDISKS_IGNORE` in the iso9660 case.

I think I prefer 2. It means we consistently always read the signature etc. from `/dev/mapper/endless-image-device`.